static/js/compose_actions.js

```diff
@@ -421,7 +445,19 @@ exports.quote_and_reply = function (opts) {
         autosize.update($('#compose-textarea'));
     }
 
-    if (message && message.raw_content) {
+    if (message && opts.selected_text !== undefined) {
+        // The copied text is plain text, therefore, the quoted text
+        // is not bold/italicized/etc.
+        message.raw_content = opts.selected_text;
+        replace_content(message);
+
+        // Resets the raw_content, otherwise this can cause issues
+        // with the `>` hotkey (the selected text may get quoted in
+        // that case).
+        message.raw_content = undefined;
```

Comment:
I don't understand why that's a bad thing. If I select some text, I think that the `>` hotkey makes more sense for quoting the selection and replying.

In fact, between `r` and `>`, the later already has quoting behavior so it feels natural to extend that instead of r.

Reply:
I've changed the this in my updated commit. The `>` hotkey now triggers this instead of the `r` hotkey.